File: api-tester/src/main/java/com/revenuecat/apitester/java/InstallmentsInfoAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.models.GoogleInstallmentsInfo;
import com.revenuecat.purchases.models.InstallmentsInfo;

@SuppressWarnings({"unused"})
final class InstallmentsInfoAPI {
    static void check(final InstallmentsInfo installmentsInfo) {
        int commitmentPaymentsCount = installmentsInfo.getCommitmentPaymentsCount();
        int subsequentCommitmentPaymentsCount = installmentsInfo.getSubsequentCommitmentPaymentsCount();
    }

    static void checkGoogleInstallmentsInfo(final GoogleInstallmentsInfo googleInstallmentsInfo) {
        check(googleInstallmentsInfo);
        int commitmentPaymentsCount = googleInstallmentsInfo.getCommitmentPaymentsCount();
        int subsequentCommitmentPaymentsCount = googleInstallmentsInfo.getSubsequentCommitmentPaymentsCount();

        GoogleInstallmentsInfo constructedGoogleInstallmentsInfo = new GoogleInstallmentsInfo(
                commitmentPaymentsCount,
                subsequentCommitmentPaymentsCount
        );
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/InstallmentsInfoAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.models.GoogleInstallmentsInfo
import com.revenuecat.purchases.models.InstallmentsInfo

@Suppress("unused", "UNUSED_VARIABLE")
private class InstallmentsInfoAPI {
    fun checkInstallmentsInfo(installmentsInfo: InstallmentsInfo) {
        val commitmentPaymentsCount: Int = installmentsInfo.commitmentPaymentsCount
        val subsequentCommitmentPaymentsCount: Int = installmentsInfo.subsequentCommitmentPaymentsCount
    }

    fun checkGoogleInstallmentsInfo(googleInstallmentsInfo: GoogleInstallmentsInfo) {
        checkInstallmentsInfo(googleInstallmentsInfo)
        val commitmentPaymentsCount: Int = googleInstallmentsInfo.commitmentPaymentsCount
        val subsequentCommitmentPaymentsCount: Int = googleInstallmentsInfo.subsequentCommitmentPaymentsCount

        val newGoogleInstallmentsInfo = GoogleInstallmentsInfo(
            commitmentPaymentsCount,
            subsequentCommitmentPaymentsCount,
        )
    }
}
